### DISCUSSION TITLE
How should I apply it to a new game

### DISCUSSION THREAD

**Discussion by zhangnan24** [BAAI-Agents/Cradle#55] [Q&A]
Taking dealers as an example, it seems that many files such as ui_control. py and basic_skills. py are specially customized based on a certain game. But how can I get these files for a new game?

**Comment by WeihaoTan** [comment]
Hi. Thanks for reaching out. Since different games have different settings and challenges,  we have to prepare some logits for the new envs first. For example, different games have different ways of pausing the game and some games even do not need to pause the game. And because of the limitations of GPT-4o's ability, we need to use some visual prompting methods to augment the image and then feed them to GPT-4o. Besides, because some games do not provide detailed guidance, we need to provide some preliminary skills first.

Here are some general steps to adapt Cradle to a new game:
1. Check whether the game needs to be paused while waiting for GPT-4o's response. If yes, implement it.
2. Check whether the game provides enough guidance for generating necessary skills (e.g., RDR2 does not teach us the usage of ASD and how to move the view). If no, implement the missing ones. Some games do not provide any guidance, then you need to implement the skills in advance, which is just like the case in web navigation. Pay attention to the specific/wired mechanisms of the games.
3. Write the prompt for each module.
4. Iteratively revise the prompt, add augmented tools and run the agent until the agent can complete the task.

**Comment by zhangnan24** [comment]
你好，你的邮件我已经收到，我将在阅读后尽快给你回复，谢谢来件∩_∩
